src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('side cannot be less than or equal to zero');
+    }
 
+    const maxSide: number = Math.max(this.a, this.b, this.c);
+    const sumTwoSide : number = this.a + this.b + this.c - maxSide;
+
+    if (maxSide >= sumTwoSide) {
+      throw new
+      Error(`sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`);
+    }
+  }
+
+  getArea() : number {
+    const s = 0.5 * (this.a + this.b + this.c);
+
+    return (Math.floor((Math.sqrt(s * (s - this.a) * (s
+      - this.b) * (s - this.c))) * 100)) / 100;
```

Comment:
You need to refactor this part, it's difficult to read. Create variable for area and pay attention to formatting(don't try to write everything in 1 line)